tests/test_util.py

```diff
@@ -456,3 +459,46 @@ def test_df_parser(tmpdir):
     assert len(df.labels) == 1
     assert df.labels.get('label') == 'foobar barfoo'
 
+def test_df_parser_parent_env_arg(tmpdir):
+    tmpdir_path = str(tmpdir.realpath())
+    p_env = {
+        "test_env": "first"
+    }
+    df_content = """\
+FROM fedora
+ENV foo=bar
+LABEL label="foobar $test_env"
+"""
+   
+    with open(tmpdir_path + '/Dockerfile', 'w') as dockerfile:
+        dockerfile.write(df_content)
+        dockerfile.flush()
+
+    df = df_parser(tmpdir_path, parent_env=p_env)
+    assert df.parent_env.get('test_env') == 'first'
```

Comment:
I think `parent_env` isn't really meant to be a public part of the interface for that object.